### PR TITLE
Follow up to #10617 and allow other `ZoneHVAC:*` component with Airflow Network simulations

### PR DESCRIPTION
**Pull request by lymereJ**
Pull request overview
---------------------
This PR is a follow up to #10617, it adds support for other types of zonal systems such as VRF terminals, PTHP, PTAC, and WSHP. The PR also address some of the review comments from #10617.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 1]
```diff
@@ -10841,6 +10841,28 @@ void getVRFTUZoneLoad(
...
+             nodeNumber == vrfTU.fanOutletNode || nodeNumber == vrfTU.heatCoilAirOutNode || nodeNumber == vrfTU.coolCoilAirOutNode ||
+             nodeNumber == vrfTU.VRFTUOAMixerOANodeNum || nodeNumber == vrfTU.VRFTUOAMixerRelNodeNum || nodeNumber == vrfTU.VRFTUOAMixerRetNodeNum ||
+             nodeNumber == vrfTU.VRFTUOAMixerMixedNodeNum || nodeNumber == vrfTU.SuppHeatCoilAirInletNode ||
+             nodeNumber == vrfTU.SuppHeatCoilAirOutletNode)) {
+            return true;
+        }
+    }
+    return false;
+}
+
```
Comment (lymereJ) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:10865]: New function to identify nodes used by VRF terminal units that don't use any OA so they can be excluded from the AFN distribution validation.

File: src/EnergyPlus/UnitarySystem.cc [hunk 2]
```diff
@@ -16739,6 +16747,33 @@ namespace UnitarySystems {
...
+                                               nodeNumber == unitarySys.m_OAMixerNodes[1] || nodeNumber == unitarySys.m_OAMixerNodes[2]) ||
+                    nodeNumber == unitarySys.m_OAMixerNodes[3] || nodeNumber == unitarySys.CoolCoilOutletNodeNum ||
+                    nodeNumber == unitarySys.HeatCoilOutletNodeNum) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
```
Comment (lymereJ) [inline src/EnergyPlus/UnitarySystem.cc:16776]: New function to identify nodes used by PTAC/PTHP/WSHP that don't use any OA so they can be excluded from the AFN distribution validation.

File: src/EnergyPlus/WindowAC.cc [hunk 3]
```diff
@@ -1499,30 +1500,22 @@ namespace WindowAC {
...
-            if (windowACOutdoorAir &&
-                (NodeNumber == windowAC.OutsideAirNode || NodeNumber == windowAC.MixedAirNode || NodeNumber == windowAC.AirReliefNode ||
-                 NodeNumber == FanInletNodeIndex || NodeNumber == FanOutletNodeIndex || NodeNumber == windowAC.AirInNode)) {
+            int FanInletNodeIndex = state.dataFans->fans(windowAC.FanIndex)->inletNodeNum;
+            int FanOutletNodeIndex = state.dataFans->fans(windowAC.FanIndex)->outletNodeNum;
+
+            if (windowAC.OutAirVolFlow == 0 &&
+                (nodeNumber == windowAC.OutsideAirNode || nodeNumber == windowAC.MixedAirNode || nodeNumber == windowAC.AirReliefNode ||
+                 nodeNumber == FanInletNodeIndex || nodeNumber == FanOutletNodeIndex || nodeNumber == windowAC.AirInNode ||
+                 nodeNumber == windowAC.CoilOutletNodeNum || nodeNumber == windowAC.AirOutNode || nodeNumber == windowAC.ReturnAirNode)) {
```
Comment (lymereJ) [inline src/EnergyPlus/WindowAC.cc:1518]: Address review comments from #10617.

File: src/EnergyPlus/UnitarySystem.cc [hunk 4]
```diff
@@ -10149,7 +10149,8 @@ namespace UnitarySystems {
             Real64 TotalOutputDelta = 0.0;    // delta total output rate, {W}
             int ZoneInNode = this->m_ZoneInletNode;
             Real64 MassFlowRate = state.dataLoopNodes->Node(ZoneInNode).MassFlowRate / this->ControlZoneMassFlowFrac;
-            if (state.afn->distribution_simulated) {
+            if (state.afn->distribution_simulated && this->m_sysType != SysType::PackagedAC && this->m_sysType != SysType::PackagedHP &&
+                this->m_sysType != SysType::PackagedWSHP) {
```
Comment (lymereJ) [inline src/EnergyPlus/UnitarySystem.cc:10153]: Exclude PTHP/PTAC/WSHP since they are excluded from AFN simulation with distribution.

**Review by lymereJ — COMMENTED** [review]
Code change walk-through.

**Review by jasondegraw — APPROVED** [review]
This appears to be following the pattern that was set out in previous PRs. I guess if I have a concern it's that some of the logical conditions are getting a little on the long side, but I'm not sure what we can do to help there. Maybe a comment on what the conditions all mean? I'm not sure about that. Anyway, no objections from me.

**Comment by Myoldmopar** [comment]
> Anyway, no objections from me.

Thanks for looking this over @jasondegraw !  This is on track to merge then.

**Comment by Myoldmopar** [comment]
This is still all happy with develop pulled in, thanks @lymereJ and @jasondegraw

**Pull request merged**